pkg/client/openshift/route.go

```diff
@@ -41,19 +41,19 @@ type route struct {
 	client *client.Client
 }
 
-func (r *route) GetHostFromRoute(routeKey types.NamespacedName) (string, error) {
+func (r *route) GetHostFromRoute(routeKey types.NamespacedName) (bool, string, error) {
 	route := &routev1.Route{}
 
 	for i := 1; i < 60; i++ {
```

Comment:
why this for loop is needed?
is there a feasible way than a loop + sleep to do it?
If my understanding is correct the route should be returned right way, yes?

Reply:
I haven't followed the full execution stack well enough to know how easy it would be, but: after creating the route, you could return with a Requeue of say, 300ms, to allow the route to likely be created and assigned a hostname. Then at this point, if no hostname is found, you could also just return with a shorter Requeue, e.g., 100ms, and let reconcile call again.

Reply:
@bmozaffa that's what we've done in the past in BA Operator, works well.
https://github.com/kiegroup/kie-cloud-operator/blob/0f35e826bf276d3f2d23fabbb266f21434bcae21/pkg/controller/kieapp/kieapp_controller.go#L104

Reply:
Thanks @spolti @bmozaffa @tchughesiv . The sleep loop is [deleted](https://github.com/kiegroup/kogito-cloud-operator/blob/33e9918d0604a0593a609031c21facc071cb286f/pkg/client/openshift/route.go#L45). If the route is not found, a [requeue](https://github.com/kiegroup/kogito-cloud-operator/blob/33e9918d0604a0593a609031c21facc071cb286f/pkg/controller/kogitoapp/status/manage_status.go#L275) will be used.